e2e/e2etests/test_solana_withdraw_restricted_address.go

```diff
@@ -0,0 +1,36 @@
+package e2etests
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/e2e/runner"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+)
+
+func TestSolanaWithdrawRestricted(r *runner.E2ERunner, args []string) {
+	require.Len(r, args, 2)
+
+	// parse restricted address
+	receiverRestricted, err := chains.DecodeSolanaWalletAddress(args[0])
+	require.NoError(r, err, fmt.Sprintf("unable to decode solana wallet address: %s", args[0]))
+
+	// parse withdraw amount (in lamports), approve amount is 1 SOL
+	approvedAmount := new(big.Int).SetUint64(solana.LAMPORTS_PER_SOL)
+	withdrawAmount := parseBigInt(r, args[1])
+	require.Equal(
+		r,
+		-1,
+		withdrawAmount.Cmp(approvedAmount),
+		"Withdrawal amount must be less than the approved amount (1e9).",
+	)
+
+	// withdraw
+	cctx := r.WithdrawSOLZRC20(receiverRestricted, withdrawAmount, approvedAmount)
```

Comment:
cctx should be reverted or aborted? Can we check the status?

Reply:
Cctx status is checked inside `WithdrawSOLZRC20` as `CctxStatus_OutboundMined`